### PR TITLE
l2geth: Use Google PubSub for transaction syncing

### PR DESCRIPTION
**Pull request by Inphi**
This reverts the pub/sub feature introduced by eedbebbd36d6a6e0a3a34460258fa479c99327db, with additional fixes. This change will go through nightly so it can be tested properly before being merged.

### CODE REVIEW

File: .changeset/rotten-dragons-poke.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (tynes) [inline .changeset/rotten-dragons-poke.md:1]: You can squash out this changeset

File: l2geth/rollup/sync_service.go [hunk 2]
```diff
@@ -869,9 +883,24 @@ func (s *SyncService) applyTransactionToTip(tx *types.Transaction) error {
 		s.SetLatestEnqueueIndex(queueIndex)
 	}
 
+	// This reset function MUST be called on any error case
+	reset := func() {
+		s.SetLatestL1Timestamp(ts)
+		s.SetLatestL1BlockNumber(bn)
+		s.SetLatestIndex(index)
+		s.SetLatestEnqueueIndex(queueIndex)
```
Comment (Inphi) [inline l2geth/rollup/sync_service.go:891]: Per @cfromknecht's [comment](https://github.com/ethereum-optimism/optimism/pull/2395#discussion_r841995467) should we reset the enqueue index?
cc @tynes 

Did we have tests for the deadlock bug?

**Comment by mslipper** [comment]
@tynes can you confirm here why we don't reset the enqueue index when the transaction fails to apply? This commit here: https://github.com/ethereum-optimism/optimism/pull/2417/commits/6f901455dd56e32580daa56991138c31503be0c5

**Comment by tynes** [comment]
> @tynes can you confirm here why we don't reset the enqueue index when the transaction fails to apply? This commit here: [6f90145](https://github.com/ethereum-optimism/optimism/commit/6f901455dd56e32580daa56991138c31503be0c5)

If a deposit causes a problem when being ingested, we don't want to halt the ingestion of all deposits by having the routine be stuck in an infinite loop. The deposit can be replayed via the contracts which will allow for it to go through. There should never be a problem when a deposit is being ingested to cause such an error

**Pull request closed without merging**